### ISSUE TITLE
efficiency improvements to `quad.3form()`

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/quadform#2]
For `quad.3form()` the order of multiplication can make a difference:

https://en.wikipedia.org/wiki/Matrix_chain_multiplication

**Comment by RobinHankin** [comment]
Function `quad.3form()` currently returns `crossprod(crossprod(M, Conj(l)), r)` (that is, multiplication by `l` and then by `r`, as in `(AB)C`.  The only ways I have found to replicate `quad.3form()` but bracketed `A(BC)` follow:

``` r
suppressMessages(library("quadform"))
suppressMessages(library("cmvnorm"))
M <- matrix(rcnorm(49),7,7)
r <- matrix(rcnorm(21),7,3)
l <- matrix(rcnorm(14),7,2)
quad.3form(M,l,r)
#>                     [,1]               [,2]                [,3]
#> [1,]  5.413950+3.843653i 1.780745+5.751373i -0.758729+2.897683i
#> [2,] -6.209812+4.257198i 2.668621+2.336866i -4.081060+1.623583i
tcrossprod(crossprod(Conj(l),M),t(r))
#>                     [,1]               [,2]                [,3]
#> [1,]  5.413950+3.843653i 1.780745+5.751373i -0.758729+2.897683i
#> [2,] -6.209812+4.257198i 2.668621+2.336866i -4.081060+1.623583i
cprod(l,M) %*% r
#>                     [,1]               [,2]                [,3]
#> [1,]  5.413950+3.843653i 1.780745+5.751373i -0.758729+2.897683i
#> [2,] -6.209812+4.257198i 2.668621+2.336866i -4.081060+1.623583i
```

Neither of them are perfect [the first has a transpose, the second uses `%*%`] but it might be worth doing some timings.

**Comment by RobinHankin** [comment]
Fixed in 01fd71e
